### PR TITLE
new formula: ODB C++ Object-Relational Mapping(ORM)

### PR DESCRIPTION
**Pull request by leezii**
Now ODB compiler is built from source.

The remain problem is I must use this command to install the formula
brew install odb --env=std since I use homebrew gcc.

Can I avoid it?

@adamv @mikemcquaid @mistydemeo

### CODE REVIEW

**Comment by leezii** [comment]
@mikemcquaid Today I push some commits, but it can't pass the brew style odb. 
The stacktrace tells me #{prefix}/lib should be #{lib}, but I don't use #{prefix}/lib. I use #[prefix]/libcutl.
Stacktrace
        ==> brew style odb
== /usr/local/Library/Formula/odb.rb ==
C: 21: 30: Avoid using {...} for multi-line blocks.
C: 26:  5: Missing space after #.
C: 37:  1: Extra empty line detected at class body end.
1 file inspected, 3 offenses detected
Error: 2 problems in 1 formulae
==> audit problems
odb:
 * "#{prefix}/lib" should be "#{lib}"
 * "#{prefix}/lib" should be "#{lib}"

File: Library/Formula/odb.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Odb < Formula
+  homepage "http://www.codesynthesis.com/products/odb"
+  url "http://www.codesynthesis.com/download/odb/2.3/odb-2.3.0.tar.gz"
+  sha1 "53c851e3f3724b72d7c7a74c497c50c195729ad1"
+
+  fails_with :clang
+  fails_with :gcc_4_0
+  fails_with :gcc
+  fails_with :llvm
```
Comment (mikemcquaid) [inline Library/Formula/odb.rb]: It fails with all of these?

File: Library/Formula/odb.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on "gcc"
+
+  resource "libcutl" do
+    url "http://www.codesynthesis.com/download/libcutl/1.8/libcutl-1.8.1.tar.gz"
+    sha1 "5411892a2959b6164321ebfb6e8e52255786b143"
+  end
+  patch do
+    url "http://codesynthesis.com/~boris/tmp/odb/odb-2.3.0-gcc-4.9.0.patch"
+    sha1 "9d0fe9db9d8667c76cd6a1bb15e911560796b656"
+  end
```
Comment (mikemcquaid) [inline Library/Formula/odb.rb]: Add a newline after each `end`

File: Library/Formula/odb.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  def install
+    resource("libcutl").stage{
+      system "./configure", "--prefix=#{prefix}/libcutl"
+      system "make", "install"
+    }
+    ENV.append "CXXFLAGS", "-fno-devirtualize -I#{prefix}/libcutl/include -L#{prefix}/libcutl/lib -I/usr/local/include"
```
Comment (mikemcquaid) [inline Library/Formula/odb.rb]: Don't add `/usr/local/include` in here; it isn't needed and you can't rely on Homebrew being installed there.

File: Library/Formula/odb.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    system "./configure", "--prefix=#{prefix}"
+    system "make", "install"
+    bin.env_script_all_files(libexec/"bin", {})
+    (libexec/"libexec/odb").dirname.mkpath
+    (libexec/"libexec").install_symlink(libexec/"odb")
+  end
+
+  test do
+    system "odb", "--help"
```
Comment (mikemcquaid) [inline Library/Formula/odb.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by mikemcquaid** [comment]
Just ignore the warning for now, might fix it later.

File: Library/Formula/odb.rb (hunk 1 above)
Comment (leezii) [inline Library/Formula/odb.rb]: It should use homebrew's gcc, so I guess it would fails with all of these.

**Pull request closed without merging**